fms2_io/netcdf_io.F90

```diff
@@ -644,7 +648,26 @@ function netcdf_file_open(fileobj, path, mode, nc_format, pelist, is_restart, do
     endif
 
     if (string_compare(mode, "read", .true.)) then
-      err = nf90_open(trim(fileobj%path), nf90_nowrite, fileobj%ncid, chunksize=fms2_ncchksz)
+      ! Open the file for collective reads if the user requested that treatment
+      ! NetCDF does not have the ability to specify collective I/O at the file basis
+      ! so we must activate at the variable level in netcdf_read_data_2d() and netcdf_read_data_3d()
+      if(fileobj%use_collective .and. fileobj%TileComm < 0) then
+        !write(6,'("netcdf_file_open: Open for collective read "A,I4)') trim(fileobj%path), szTile
+        err = nf90_open(trim(fileobj%path), ior(NF90_NOWRITE, NF90_MPIIO), fileobj%ncid, comm=fileobj%TileComm, info=MPI_INFO_NULL)
+        if(err /= nf90_noerr) then
+          err = nf90_open(trim(fileobj%path), nf90_nowrite, fileobj%ncid)
+          err = nf90_get_att(fileobj%ncid, nf90_global, "_IsNetcdf4", IsNetcdf4)
+          err = nf90_close(fileobj%ncid)
+          if(IsNetcdf4 /= 1) then
+            write(6,'("netcdf_file_open: Open for collective read failed because the file is not netCDF-4 format. &
```

Comment:
If this is going to stay in the code, it should use `mpp_error` with `NOTE`

Reply:
I will change the code to use mpp_error with a NOTE.